### PR TITLE
New: Add Subsonic library update and notify

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
NO

#### Description
Adds Subsonic as a connection option and adds ability to update Subsonic library or Notify subsonic via Chat Message

#### Todos
- [x] Documentation

#### Issues Fixed or Closed by this PR
https://feathub.com/lidarr/Lidarr/+26

### CODE REVIEW

File: src/NzbDrone.Core/Notifications/Subsonic/SubsonicServerProxy.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            {
+                throw new SubsonicException("Could not read version from Subsonic");
+            }
+
+            return version;
+        }
+
+        private RestClient GetSubsonicServerClient(SubsonicSettings settings)
+        {
+            var protocol = "http";
```
Comment (danielunderwood) [inline src/NzbDrone.Core/Notifications/Subsonic/SubsonicServerProxy.cs]: Should we have an option to use https?

File: src/NzbDrone.Core/Notifications/Subsonic/SubsonicServerProxy.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        private RestRequest GetSubsonicServerRequest(string resource, Method method, SubsonicSettings settings)
+        {
+            var request = new RestRequest(resource, method);
+
+            if (settings.Username.IsNotNullOrWhiteSpace())
+            {
+                request.AddParameter("u", settings.Username);
+                request.AddParameter("p", settings.Password);
+                request.AddParameter("c", "Lidarr");
+                request.AddParameter("v", "1.15.0");
```
Comment (danielunderwood) [inline src/NzbDrone.Core/Notifications/Subsonic/SubsonicServerProxy.cs]: What's this version for? Is it going to stay at this version?

Comment (Qstick) [reply src/NzbDrone.Core/Notifications/Subsonic/SubsonicServerProxy.cs]: That defines what min version the server has to for our call to process. They just added the `startScan` api route in 1.15.0. http://www.subsonic.org/pages/api.jsp#startScan

File: src/NzbDrone.Core/Notifications/Subsonic/SubsonicServerProxy.cs (hunk 1 above)
Comment (Qstick) [reply src/NzbDrone.Core/Notifications/Subsonic/SubsonicServerProxy.cs]: Oops yes we should.

**Comment by Qstick** [comment]
Add setting for SSL and add wiki docs: https://github.com/lidarr/Lidarr/wiki/Supported-Notifications#subsonic

**Comment by Qstick** [comment]
I’ve installed and been testing some today, seems to work as expected. Going to merge for nightly testing.

**Pull request merged**